src/cr/cube/matrix/assembler.py

```diff
@@ -692,6 +771,88 @@ def _assemble_matrix(self, blocks):
         # --- desired output.
         return np.block(blocks)[np.ix_(self._row_order, self._column_order)]
 
+    def _assemble_pw_indices(self, p_vals, t_stats, alpha, only_larger):
+        """Return assembled 2D ndarray ot tuple of pairwise int indices.
+
+        Significance indices are calculated iterating over each ordered pvals and
+        t_stats column and assembled into a 2D array.
+
+        This assembler method generates a 2D array of tuples of integers starting from
+        t_stats and p_vals significance matrices:
+        pvals:
+        [[
+           [0.1  0.2  0.3 ]    pvals_col_0_selected
+           [0.4  0.5  0.6 ]
+         ]
+         [
+           [0.01 0.02 0.3 ]    pvals_col_1_selected
+           [0.02 0.03 0.04]
+         ]]
+
+        t_stats:
+        [[
+           [0.1  0.2  0.3 ]    t_stats_col_0_selected
+           [0.4  0.5  0.6 ]
+         ]
+         [
+           [0.01 0.02 0.3 ]    t_stats_col_1_selected
+           [0.02 0.03 0.04]
+         ]]
+        Each pvals and tstats matrix is used to calculate the 1D array of tuple of int
+        that represents the significance for the selected column:
+        e.g. [(0,),(), (1,2)]
+        Iterating over each significance matrix the method assemble the complete matrix
+        of significant indices:
+        [[(0,), (), (1,2)],
+         [(), (), (0,1]]
+        """
+
+        def pairwise_indices(p_vals, t_stats):
+            """1D ndarray containing tuples of int pairwise indices of each column."""
+            significance = p_vals < alpha
+            if only_larger:
+                significance = np.logical_and(t_stats < 0, significance)
+            col_significance = np.empty((len(significance),), dtype=object)
+            col_significance[:] = [
+                tuple(np.where(sig_row)[0]) for sig_row in significance
+            ]
+            return col_significance
+
+        return np.array(
+            [
+                pairwise_indices(
+                    p_vals[i][np.ix_(self._row_order, self._column_order)],
+                    t_stats[i][np.ix_(self._row_order, self._column_order)],
+                )
+                for i in self._column_order
+            ]
+        ).T
+
+    def _assemble_pw_scale_mean_indices(self, p_vals, t_stats, alpha, only_larger):
+        """List of tuples indicating the significance of scale means values
+
+        Considering this output: [(3,), (2, 3), (3,), ()] and scale_means values
+        [26, 30, 21, 11], each element contains a tuple of other element
```

Comment:
This all doesn't seem right to me. Measures should not be computed in assembler. That's what the measure is for. Otherwise the measure isn't available to other measures for secondary computations.

The idea would be to compute the _payload order_ indices and then just translate those payload column-order indices to sorted column-order indices in assembler. The rest of assembling the matrix should work fine just like it does for any other measure and you don't need to reimplement it here.

Basically you construct a mapping of payload order to sorted order, which is very much like if not exactly the `._column_order` value. Then you visit each tuple in the indices array and exchange each integer in each tuple with the _sorted-order_ index instead of the payload-order index is started with. Then you're done. Everything else happens the same way as any other measure.

Reply:
ok I'll try to map the indices